webui/src/util.jsx

```diff
@@ -29,6 +29,32 @@ export function socketEmit(socket, name, args, timeout, timeoutMessage) {
 	return pTimeout(p, timeout, timeoutMessage ?? `Timed out after ${timeout / 1000}s`)
 }
 
+export function sandbox(serializedFn) {
```

Comment:
I was curious about this, as it looks great but feels too simple.

Came across [this article](https://portswigger.net/research/attacking-and-defending-javascript-sandboxes) which has some criticisms of the original implemenation and suggests some improvements. It would be good to pull those in

This feels like something that there should be an npm packge for, but a quick search didnt come up with one

Reply:
@Julusian if one doesn't exist should the project deploy one as such?

Reply:
> Came across this article which has some criticisms of the original implemenation and suggests some improvements. It would be good to pull those in
I'll check the article to see if there's anything I can improve.

> @Julusian if one doesn't exist should the project deploy one as such?
Yes I intended to eventually bundle it up into it's own npm package.

Reply:
That article is GREAT! I need to incorporate some of those features into our sandbox function.

Reply:
> > @Julusian if one doesn't exist should the project deploy one as such?
> 
> Yes I intended to eventually bundle it up into it's own npm package.

I'm not sure if it is worth it, as it means that making any changes will be harder. Unless @estilles wants to reuse it in other things, then it would be worth it

Reply:
I will probably create a more general use version of this in the figure and turn it into an npm module, but this one is a very customized for our use case.

Reply:
@Julusian ... based on the article you pointed out (thanks by the way) I've added the following:

- allow specific global objects
- freeze allowed global objects to prevent `indexOf` attack
- use a double proxy approach
- add `"use strict"` to prevent `window` object leaking
- protect against "aysnc functions/generators" bypassing sandbox